src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -60,7 +60,8 @@ public void EndStreamValueScope()
         /// <returns>A task representing the asynchronous operation. The task result contains a stream for writing the stream value.</returns>
         public async Task<Stream> StartStreamValueScopeAsync()
         {
-            this.WriteSeparatorIfNecessary();
+            await this.utf8JsonWriter.FlushAsync();
+            this.WriteItemWithSeparatorIfNeeded();

```

Comment:
I think it's wasteful to flush twice in this method. We should use `this.CommitUtf8JsonWrtierContentsToBuffer` like we do in the sync version. We only need to make sure that the buffer writer's position is advanced passed the Utf8JsonWriter's content before we write to the buffer directly, we don't need a full-blown flush yet.